source/guide_puma.rst

```diff
@@ -0,0 +1,197 @@
+.. highlight:: console
+
+.. sidebar:: Logo
+  
+  .. image:: _static/images/puma.png
+      :align: center
+
+#######
+Puma
+#######
+
+Puma_ is a Ruby_ web server built for speed and parallelism. It is designed for running Rack_ apps only. It is licensed under the BSD 3-Clause license. This guide explains how to install Puma and run a minimal, custom Ruby application.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of 
+
+  * Ruby_
+  * supervisord_
+
+Installation
+============
+
+Use ``gem`` to install the latest version of Puma:
+
+::
+
+  [isabell@stardust ~]$ gem install puma
+  Fetching: puma-3.12.0.gem (100%)
+  Building native extensions. This could take a while...
+  Successfully installed puma-3.12.0
+  1 gem installed
+  [isabell@stardust ~]$ 
+  
+Create a directory, and inside the directory a file ``rubyapp.ru`` for the application:
+
+::
+
+  [isabell@stardust ~]$ mkdir ~/rubyapp
+  [isabell@stardust ~]$ touch ~/rubyapp/rubyapp.ru
+  [isabell@stardust ~]$ 
+
+Update the permissions so that the application file is executable:
+
+::
+
+  [isabell@stardust ~]$ chmod +x ~/rubyapp/rubyapp.ru 
+  [isabell@stardust ~]$ 
+  
+This file is the Rackup file for the application. Add the following content to ``~/rubyapp/rubyapp.ru``, a simple Hello World application:
+
+.. code-block:: none
+  class HelloWorld
+    def call(env)
+      [200, {"Content-Type" => "text/plain"}, ["Hello World"]]
+    end
+  end
+
+  run HelloWorld.new
+
+
+Configuration
+=============
+
+Configure port
+--------------
+
+You need to find a free port to bind Puma to it.
+
+.. include:: includes/generate-port.rst
+
+Puma configuration file
+-----------------------
+
+Next, create a Puma configuration file, update the permissions as before,
+
+::
+
+  [isabell@stardust ~]$ touch ~/rubyapp/config.rb
+  [isabell@stardust ~]$ chmod +x ~/rubyapp/config.rb
+  [isabell@stardust ~]$
+
+and add the following content:
+
+.. code-block:: none
+  #!/usr/bin/env puma
+
+  # The directory to operate out of.
+  directory '/home/isabell/rubyapp'
+
+  # The path to the rackup file
+  rackup '/home/isabell/rubyapp/rubyapp.ru'
+
+  # Bind Puma to the port
+  bind 'tcp://127.0.0.1:9000'
+
+
+Setup supervisord
+-----------------
+
+Create ``~/etc/services.d/rubyapp.ini`` with the following content:
+
+.. code-block:: ini
+  [program:rubyapp]
+  command=/opt/uberspace/etc/isabell/binpaths/ruby/puma --config /home/isabell/rubyapp/config.rb
+  autostart=yes
+  autorestart=yes
+
+The ``--config`` parameter provides the path to the configuration file.
+
+Tell ``supervisord`` to refresh its configuration and start the service:
+
+::
+
+  [isabell@stardust ~]$ supervisorctl reread
+  rubyapp: available
+  [isabell@stardust ~]$ supervisorctl update
+  rubyapp: added process group
+  [isabell@stardust ~]$ 
+
+Setup .htaccess
+---------------
+
+Create the file ``~/html/.htaccess`` with the following content to forward requests from the outside to Puma:
```

Comment:
Users will just copy and paste your code. Mention that the port needs to be changed and work with an example as above.